lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py

```diff
@@ -531,7 +531,12 @@ def main():
             host_param = module.params['host']
             if not host_param:
                 host = search_by_attributes(connection.system_service().hosts_service(), status='up')
-                host_param = host.name if host is not None else None
+                if host is None:
```

Comment:
What does engine return in that case? I believe engine should send appropriate error.

Reply:
Note that here we are in the case that user did NOT fill the host param. Yet we are supposed to auto-fill the host parameter it with an available host that is up.

In case there is not available host, we fill in the code the host parameter with None here: 
https://github.com/ansible/ansible/blob/2caae251ff18ca7b14ef2a8fbbf820ce529e32ae/lib/ansible/modules/cloud/ovirt/ovirt_storage_domains.py#L534
, and then it failed, with 'host parameter is missing' error.

I don't see how engine can react different here. You have something else in mind?